jacodb-panda-dynamic/src/main/kotlin/org/jacodb/panda/dynamic/ets/dto/convert.kt

```diff
@@ -0,0 +1,712 @@
+                field = field
+            )
+        }
+    }
+
+    fun cfg2cfg(cfg: CfgDto): EtsCfg {
+        require(cfg.blocks.isNotEmpty()) {
+            "Method body should contain at least return stmt"
+        }
+
+        val visited: MutableSet<Int> = hashSetOf()
+        val queue: ArrayDeque<Int> = ArrayDeque()
+        queue.add(0)
+
+        val blocks = cfg.blocks.associateBy { it.id }
+        val blockStart: MutableMap<Int, Int> = hashMapOf()
+        val blockEnd: MutableMap<Int, Int> = hashMapOf()
+
+        while (queue.isNotEmpty()) {
+            val block = blocks[queue.removeFirst()]!!
+            if (block.stmts.isNotEmpty()) {
+                blockStart[block.id] = currentStmts.size
+            }
+            for (stmt in block.stmts) {
+                convertToEtsStmt(stmt)
+            }
+            if (block.stmts.isNotEmpty()) {
+                blockEnd[block.id] = currentStmts.lastIndex
+            }
+            for (next in block.successors) {
+                if (visited.add(next)) {
+                    queue.addLast(next)
+                }
+            }
+        }
+
+        val successorMap: MutableMap<EtsStmt, List<EtsStmt>> = hashMapOf()
+        for (block in cfg.blocks) {
+            if (block.stmts.isEmpty()) {
+                continue
+            }
+            val startId = blockStart[block.id]!!
+            val endId = blockEnd[block.id]!!
+            for (i in startId until endId) {
+                successorMap[currentStmts[i]] = listOf(currentStmts[i + 1])
+            }
+            successorMap[currentStmts[endId]] = block.successors.mapNotNull { blockId ->
+                blockStart[blockId]?.let { currentStmts[it] }
+            }
+        }
+
+        return EtsCfg(
+            stmts = currentStmts,
+            successorMap = successorMap,
+        )
+    }
+}
+
+fun convertToEtsClass(classDto: ClassDto): EtsClass {
+    fun defaultConstructorDto(classSignatureDto: ClassSignatureDto) = MethodDto(
+        signature = MethodSignatureDto(
+            enclosingClass = classSignatureDto,
+            name = "constructor",
+            parameters = emptyList(),
+            returnType = ClassTypeDto(classSignatureDto)
+        ),
+        modifiers = emptyList(),
+        typeParameters = emptyList(),
+        body = BodyDto(
+            locals = emptyList(),
+            cfg = CfgDto(
+                blocks = listOf(
+                    BasicBlockDto(
+                        id = 0,
+                        successors = emptyList(),
+                        predecessors = emptyList(),
+                        stmts = listOf(
+                            ReturnStmtDto(
+                                arg = ThisRefDto(
+                                    type = ClassTypeDto(classSignatureDto)
+                                )
+                            )
+                        )
+                    )
+                )
+            )
+        )
+    )
+
```

Comment:
Please, construct these objects separately and create without soo many nested levels of instructions